### ISSUE TITLE
Buffer deprecation warning - Bump urllib to at least 3.0.0

### ISSUE THREAD

**Issue by blazejkustra** [snowflakedb/snowflake-connector-nodejs#363]
### Issue
I keep getting deprecations errors whenever I use `connection.execute(...)` function from `snowflake-connector-nodejs`.

```
DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
    at showFlaggedDeprecation (node:buffer:187:11)
    at new Buffer (node:buffer:271:3)
    at node_modules/.pnpm/formstream@1.1.1/node_modules/formstream/lib/formstream.js
    at node_modules/.pnpm/urllib@2.38.1/node_modules/urllib/lib/urllib.js 
    at node_modules/.pnpm/urllib@2.38.1/node_modules/urllib/lib/index.js 
    at NodeHttpClient.sendRequest2 
    at Gzip.cb 
```

### Explanation:
`snowflake-connector-nodejs` has `urllib@2.38.0` dependency. `urllib@2.38.0` is using `formstream@1.1.1` library under the hood which triggers deprecation error when I use `connection.execute(...)`.

In fact I use yet another library on top (`snowflake-promise`) that uses `snowflake-connector-nodejs` to execute snowflake query:
```ts
import { Snowflake } from 'snowflake-promise';
const snowflake = new Snowflake({ ... });
snowflake.connect();
await snowflake.execute(SQL) // this triggers the deprecation warning (it uses `connection.execute(...)` under the hood)
```

### What needs to be done

Update `urllib` package to version `3.0.0` - `3.5.0` as they are no longer using `formstream@1.1.1` in their implementation, therefore no deprecation warning will be triggered when using your library.

**Comment by sfc-gh-dszmolka** [comment]
thank you for submitting this issue and the additional information! we'll take a look.

**Comment by sfc-gh-dszmolka** [comment]
@blazejkustra thank you for the detailed explanation. this should be resolved now with `formstream 1.2.0`, please see the details in #357 . closing this issue as a duplicate of the older one.
